src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,197 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.servicenow.sink.service;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.cdap.plugin.servicenow.restapi.RestAPIResponse;
+import io.cdap.plugin.servicenow.sink.ServiceNowSinkConfig;
+import io.cdap.plugin.servicenow.sink.model.RestRequest;
+import io.cdap.plugin.servicenow.sink.model.ServiceNowBatchRequest;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIRequestBuilder;
+import org.apache.http.Header;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation class for ServiceNow Batch Rest API.
+ */
+public class ServiceNowSinkAPIRequestImpl {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowSinkAPIRequestImpl.class);
+  private static final String INSERT_TABLE_API_URL_TEMPLATE = "/api/now/table/%s";
+  private static final String UPDATE_TABLE_API_URL_TEMPLATE = "/api/now/table/%s/%s";
+  private static final String HTTP_POST = "POST";
+  private static final String HTTP_PUT = "PUT";
+  private static final String SYS_ID = "sys_id";
+  private static final String SERVICED_REQUESTS = "serviced_requests";
+  private static final String UNSERVICED_REQUESTS = "unserviced_requests";
+  private static Integer counter = 1;
+  private static Integer batchRequestIdCounter = 1;
+  private static Integer retryCounter = 0;
+  private final ServiceNowSinkConfig config;
+  private final ServiceNowTableAPIClientImpl restApi;
+
+  public ServiceNowSinkAPIRequestImpl(ServiceNowSinkConfig conf) {
+    this.config = conf;
+    restApi = new ServiceNowTableAPIClientImpl(config);
+  }
+
+  public RestRequest getRestRequest(JsonObject jsonObject) {
+    JsonElement jsonElement = jsonObject.get(SYS_ID);
+    String sysId = null;
+
+    if (jsonElement == null) {
+      if (config.getOperation().equals("update")) {
+        throw new RuntimeException("No sys_id found in the record to be updated");
+      }
+    } else {
+      if (config.getOperation().equals("insert")) {
+        jsonObject.remove(SYS_ID);
+      }
+      sysId = jsonElement.getAsString();
```

Comment:
can we use jsonObject.asText or jsonObject.getAsString method instead of toString